### PR TITLE
banked dcache

### PR DESCRIPTION
**Pull request by bkorpan**
verilator/vcs sim compilation tends to hang on 128-bit bus configs for some reason

### CODE REVIEW

**Comment by jerryz123** [comment]
I've noticed that too. MegaBoom dual-mem takes 2 hours for VCS compile.

@bkorpan How did you find that VecInit.tabulate was causing hangs?

**Comment by bkorpan** [comment]
> I've noticed that too. MegaBoom dual-mem takes 2 hours for VCS compile.

I fought with it for days thinking it a bug. Does the VCS step eventually finish? The part it was hanging on normally takes a couple seconds.

**Comment by jerryz123** [comment]
> > I've noticed that too. MegaBoom dual-mem takes 2 hours for VCS compile.
> 
> I fought with it for days thinking it a bug. Does the VCS step eventually finish? The part it was hanging on normally takes a couple seconds.

It does eventually finish. I have no idea exactly what might be blowing up the compile time

File: src/main/scala/common/parameters.scala [hunk 1]
```diff
@@ -67,6 +67,7 @@ case class BoomCoreParams(
   nPerfCounters: Int = 0,
   numRXQEntries: Int = 4,
   numRCQEntries: Int = 8,
+  numDCacheBanks: Int = 1,
```
Comment (jerryz123) [inline src/main/scala/common/parameters.scala:70]: What is the behavior when `numDCacheBanks=1`, `memWidth=2`. Does the load to the second pipe slot always get nack'd?

Comment (bkorpan) [reply src/main/scala/common/parameters.scala:70]: `require (nBanks >= memWidth)`

**Pull request merged**